mrp_production_real_cost/models/mrp_production.py

```diff
@@ -31,8 +31,11 @@ def _compute_real_cost(self):
     @api.multi
     def action_production_end(self):
         res = super(MrpProduction, self).action_production_end()
-        self.mapped('move_created_ids2').filtered(
-            lambda l: l.state == 'done').product_price_update_production_done()
+        for production in self:
+            production.sudo().refresh()
```

Comment:
This is needed because commit https://github.com/odoo/odoo/commit/6f29bfc181d23d70d29776d96b4318e9ee2c93a9            introduces a weird behavior on the next call, provoking an error.